pkg/util/shell/shell_test.go

```diff
@@ -0,0 +1,217 @@
+	expectedHint string
+}
+
+type prefixSuffixDelimiterTestCase struct {
+	shellName string
+	prefix    string
+	suffix    string
+	pathVar   bool
+	delimiter string
+}
+
+func TestUnknownShell(t *testing.T) {
+	expectedShellName := "foo"
+	defer func(shell string) { os.Setenv("SHELL", shell) }(os.Getenv("SHELL"))
+	os.Setenv("SHELL", "/bin/"+expectedShellName)
+
+	shell, err := GetShell("")
+	if err != nil {
+		t.Fatal("Unexpected err : ", err.Error())
+	}
+
+	if shell != expectedShellName {
+		t.Fatalf("Expected shell to be %s but %s", expectedShellName, shell)
+	}
+}
+
+func TestKnownShell(t *testing.T) {
+	expectedShellName := "bash"
+	shell, err := GetShell("bash")
+	if err != nil {
+		t.Fatal("Unexpected err : ", err.Error())
+	}
+
+	if shell != expectedShellName {
+		t.Fatalf("Expected shell to be %s but %s", expectedShellName, shell)
+	}
+}
+
+func TestNoProxyFromEnvWhenUsedLowerCase(t *testing.T) {
+	expectedNoProxyVarName := "no_proxy"
+	expectedNoProxyVarValue := "FOO_BAR"
+	defer func(val string) { os.Setenv(expectedNoProxyVarName, val) }(os.Getenv(expectedNoProxyVarName))
+	os.Setenv(expectedNoProxyVarName, expectedNoProxyVarValue)
+
+	_, noProxyValue := FindNoProxyFromEnv()
+	if noProxyValue != expectedNoProxyVarValue {
+		t.Fatalf("Expected no proxy var value as %s but got %s", expectedNoProxyVarValue, noProxyValue)
+	}
+}
+
+func TestNoProxyFromEnvWhenUsedUpperCase(t *testing.T) {
+	expectedNoProxyVarName := "NO_PROXY"
+	expectedNoProxyVarValue := "FOO_BAR"
+	defer func(val string) { os.Setenv(expectedNoProxyVarName, val) }(os.Getenv(expectedNoProxyVarName))
+	os.Setenv(expectedNoProxyVarName, expectedNoProxyVarValue)
+
+	_, noProxyValue := FindNoProxyFromEnv()
+	if noProxyValue != expectedNoProxyVarValue {
+		t.Fatalf("Expected no proxy var value as %s but got %s", expectedNoProxyVarValue, noProxyValue)
+	}
+}
+
+func TestGenerateUsageHint(t *testing.T) {
+	testData := []shellTestCase{
+		{
+			name: "bash", cmdLine: "foo", expectedHint: `# Run this command to configure your shell:
+# eval $(foo)
+`,
+		},
+		{
+			name: "fish", cmdLine: "foo", expectedHint: `# Run this command to configure your shell:
+# eval (foo)
+`,
+		},
+		{
+			name: "powershell", cmdLine: "foo", expectedHint: `# Run this command to configure your shell:
+# & foo | Invoke-Expression
+`,
+		},
+		{
+			name: "cmd", cmdLine: "foo", expectedHint: `REM Run this command to configure your shell:
+REM 	@FOR /f "tokens=*" %i IN ('foo') DO @call %i
+`,
+		},
+		{
+			name: "emacs", cmdLine: "foo", expectedHint: `;; Run this command to configure your shell:
+;; (with-temp-buffer (shell-command "foo" (current-buffer)) (eval-buffer))
+`,
+		},
+	}
+
+	for _, tt := range testData {
+		hint := GenerateUsageHint(tt.name, tt.cmdLine)
+		if tt.expectedHint != hint {
+			t.Fatalf("Expected hint to be \n%s but got \n%s", tt.expectedHint, hint)
+		}
+
+	}
+}
+
+func TestGetPrefixSuffixDelimiterForSet(t *testing.T) {
+	testData := []prefixSuffixDelimiterTestCase{
```

Comment:
I find this test not terribly useful. You are effectively just repeating what's setup in GetPrefixSuffixDelimiterForSet. IMO we can safely remove this test.